lib/staggered_quark_smearing.cu

```diff
@@ -0,0 +1,223 @@
+#include <dslash.h>
+#include <worker.h>
+#include <dslash_helper.cuh>
+#include <color_spinor_field_order.h>
+#include <gauge_field_order.h>
+#include <color_spinor.h>
+#include <dslash_helper.cuh>
+#include <index_helper.cuh>
+#include <gauge_field.h>
+#include <uint_to_char.h>
+
+#include <dslash_policy.cuh>
+#include <kernels/staggered_quark_smearing.cuh>
+
+/**
+   This is the staggered (two-link) Laplacian derivative.
+*/
+
+namespace quda
+{
+
+  template <typename Arg> class StaggeredQSmear : public Dslash<staggered_qsmear, Arg>
+  {
+    using Dslash = Dslash<staggered_qsmear, Arg>;
+    using Dslash::arg;
+    using Dslash::in;
+
+  public:
+    StaggeredQSmear(Arg &arg, const ColorSpinorField &out, const ColorSpinorField &in) : Dslash(arg, out, in) { }
+
+    void apply(const qudaStream_t &stream)
+    {
+      TuneParam tp = tuneLaunch(*this, getTuning(), getVerbosity());
```

Comment:
While this isn't going to break correctness, when `t0 == -1` on rank 0 the tuned values for the dslash part of this are going to end up very meaningless, since the kernel isn't doing any work...

Reply:
Thanks. I (or we) didn't think about it... Now, I am concerned.. What would happen when the second source's time slice is on a different rank from the first source's?

Reply:
I just had a chance to look into this, and I share @weinbe2's concern here.  I'm not sure what the solution is to this.  Perhaps we could have an interface where we can override which rank does the kernel tuning and broadcasts the tune cache from?  Then here we'd set this to being the first rank with the time slice in it.